source/vecs/entity.d

```diff
@@ -922,20 +957,14 @@ private:
 	}
 
 
-	/// Common logic for set dependencies
-	Component* _set(Component)(in Entity entity, Component component = Component.init)
-		if (isComponent!Component)
-	{
-		// set the component to entity
-		return _assureStorage!Component().set(entity, component);
-	}
-
-
-	/// Common logic for remove dependencies
-	void _remove(Component)(in Entity entity)
-		if (isComponent!Component)
+	// FIXME: documentation
+	@safe pure nothrow @nogc
+	void releaseId(in Entity e, in size_t batch)
+		in (valid(e))
+		in (batch <= Entity.maxbatch)
```

Comment:
is this intended? On `storage.d:475` this is done with `&&`. I would suggest changing to one or another.

Reply:
I'll change the other, the plan is to also add a message to each contract. If separated, I can output better messages.

Reply:
> I'll change the other, the plan is to also add a message to each contract. If separated, I can output better messages.

Yes. I prefer that too. It also appears different on the stack trace (such as line numbers)

Reply:
I read the message again. I did not read the `storage` part the first time. That case has to be with `&&` because the right side of the condition depends on the left side. I'm dealing with an array, so to guarantee I can access it, I first need to pass other checks. If those are separated in individual contracts, it will break because the condition is as a whole.

The check is to verify that there is not an entity stored with the same `id` and different `batch`.